### PR TITLE
Config with pure python

### PR DESCRIPTION
**Pull request by icemac**
* Lint the code.
* Add support for Python 3.8 and 3.9.

### CODE REVIEW

**Comment by icemac** [comment]
The drop in coverage results from the switch to branch coverage.

File: src/zope/datetime/__init__.py [hunk 1]
```diff
@@ -896,7 +907,10 @@ def _parse(self, string, local=True):
     def _validDate(self, y, m, d):
         if m < 1 or m > 12 or y < 0 or d < 1 or d > 31:
             return 0
-        return d <= self._month_len[(y % 4 == 0 and (y % 100 != 0 or y % 400 == 0))][m]
+        return d <= self._month_len[(y %
+                                     4 == 0 and (y %
+                                                 100 != 0 or y %
+                                                 400 == 0))][m]
```
Comment (mgedmin) [inline src/zope/datetime/__init__.py:913]: Oh dear, how about

```suggestion
        is_leap_year = y % 4 == 0 and (y % 100 != 0 or y % 400 == 0)
        return d <= self._month_len[is_leap_year][m]
```

**Review by mgedmin — APPROVED** [review]
LGTM

**Comment by icemac** [comment]
Thank you for reviewing this PR. 😃

**Pull request merged**